### DISCUSSION TITLE
Using Doctrine DBAL

### DISCUSSION THREAD

**Discussion by yguedidi** [FreshRSS/FreshRSS#4318] [General]
Hello!
Have you considered using Doctrine DBAL?
I'm not talking here about the ORM, just the DBAL, the abstraction layer.
I found this old comment https://github.com/FreshRSS/FreshRSS/issues/956#issuecomment-305970683 but it seems more against Doctrine ORM.
With DBAL we continue to write pure SQL :)
Thanks

**Comment by Alkarex** [comment]
Hello and thanks for the suggestion.
SQL queries are really an essential part of FreshRSS, which I would like to be optimised as much as possible.
I do not believe such an abstraction will allow expressing some of the more complex SQL queries we have.
Furthermore, SQL is already an abstraction layer :-)

**Reply by yguedidi** [reply]
It's not about the SQL, it's about all the DB connection code, PDO abstraction, etc that is home made and should be replaced by a robust known dependency
